Added ghosttrap datum, added attack_ghost() calls to posis and plants.

MistakeNot4892 opened the pull request:
See changelog. Ghosts can now harvest mob plants and enter brains, and they use a unified/modular system to get players. Harvest/posiclick now prints a message to all suitable ghosts with a href link instead of a nasty popup.

EDIT: Have tested all uses of this system, it's all working fine.

Techhead0 commented on code/modules/mob/living/carbon/brain/posibrain.dm:
```diff
@@ -55,33 +49,13 @@
-
-	src.searching = 0
+/obj/item/device/mmi/digital/posibrain/proc/reset_search()
+	if(searching || (src.brainmob && src.brainmob.key))
```
This should be `if(!searching || (src.brainmob && src.brainmob.key)`, shouldn't it?

Zuhayr replied:
Whoops, yes.

GinjaNinja32 commented on code/modules/ghosttrap/trap.dm:
```diff
@@ -0,0 +1,105 @@
+		var/mob/target = locate(href_list["target"])                     // So much BYOND magic.
+		if(!target || !candidate)
+			return
+		if(assess_candidate(candidate) && !target.ckey)
```
Should probably have a sanity check that `candidate == usr`, or potentially just remove `candidate` and use `usr`.

GinjaNinja32 commented on code/modules/ghosttrap/trap.dm:
```diff
@@ -0,0 +1,105 @@
+// Handles a response to request_player().
+/datum/ghosttrap/Topic(href, href_list)
+	if(..())
+		return
```
`return 1` here, even if it's not likely to matter.

GinjaNinja32 commented on code/modules/ghosttrap/trap.dm:
```diff
@@ -0,0 +1,105 @@
+			return
+		if(assess_candidate(candidate) && !target.ckey)
+			transfer_personality(candidate,target)
+
```
Should probably return 1 here too.

PsiOmegaDelta commented on code/modules/ghosttrap/trap.dm:
```diff
@@ -0,0 +1,114 @@
+		return
+	for(var/mob/dead/observer/O in player_list)
+		if(!O.MayRespawn())
+			continue
```
If you use ``MayRespawn(1)`` it'll arrange feedback for you.

Zuhayr replied:
I don't want feedback for this, it prints to all ghosts.

PsiOmegaDelta replied:
Ah, yes.

MistakeNot4892 committed:
Adds ghost click to posibrains.
Added ghost trap system used by plants and posibrains to get players. Added attack_ghost() to hydro trays and posibrains for the aforementioned.
Changelog.
Minor typo.
Topic fixes.
Minor fix.

MistakeNot4892 committed:
Merge branch 'dev' of https://github.com/Baystation12/Baystation12 into posibrains
